proposals/DOM-Parts.md

```diff
@@ -0,0 +1,521 @@
+or its [properties]((https://tc39.es/ecma262/#sec-object-type))
+Instead, the new value is staged to be later committed in a batch,
+which is defined by an ordered set of *DOM parts* called **DOM parts group**.
+Each *DOM part* may belong to at most one *DOM part group*.
+This batching reduces the runtime overhead of constantly returning control back from browser’s implementation to JavaScript
+between each DOM mutation and allows browser engine’s to avoid or batch certain sanity checks and housekeeping tasks.
+
+In the [old proposal](https://github.com/w3c/webcomponents/blob/gh-pages/proposals/Template-Instantiation.md#3-proposal),
+a *DOM part group* was represented by a [template element](https://html.spec.whatwg.org/multipage/scripting.html#the-template-element).
+This proposal keeps that approach possible but does not keep it the sole approach possible.
+
+### Interfaces
+
+```webidl
+interface Part {
+    attribute any value;
+    void commit();
+};
+
+interface NodePart : Part {
+    readonly attribute Node node;
+};
+
+interface AttributePart : Part {
+    constructor(Element element, DOMString qualifiedName, DOMString? namespace);
+    readonly attribute DOMString prefix;
+    readonly attribute DOMString localName;
+    readonly attribute DOMString namespaceURI;
+};
+
+interface ChildNodePart : Part {
+    constructor(Node node, Node? previousSibling, Node? nextSibling);
+    readonly attribute Node parentNode;
+    readonly attribute Node? previousSibling;
+    readonly attribute Node? nextSibling;
+};
+```
+
+In the most basic level, this proposal consists of two *DOM parts*:
+`AttributePart` represents a single [attribute](https://dom.spec.whatwg.org/#concept-attribute) which can be set
+and `ChildNodePart` represents a sequence of [child](https://dom.spec.whatwg.org/#concept-tree-child) [nodes](https://dom.spec.whatwg.org/#concept-node)
+of a [node](https://dom.spec.whatwg.org/#concept-node) which can be [replaced](https://dom.spec.whatwg.org/#concept-node-replace).
+
+### Basic Examples
+
+Suppose we had the following template where `{name}` and `{email}` are locations of our interests:
+```html
+<section><h1 id="name">{name}</h1>Email: <a id="link" href="mailto:{email}">{email}</a></section>
+```
+
+And we've parsed the following HTML so far (future extensions will make this parsing possible without extra JavaScript code):
+```html
+<section><h1 id="name"></h1>Email: <a id="link" href=""></a></section>
+```
+
+Then we can create `ChildNodePart` for [`h1` element](https://html.spec.whatwg.org/multipage/sections.html#the-h1,-h2,-h3,-h4,-h5,-and-h6-elements)
+and [`a` element](https://html.spec.whatwg.org/multipage/text-level-semantics.html#the-a-element)
+and `AttributePart` for [`a` element](https://html.spec.whatwg.org/multipage/text-level-semantics.html#the-a-element) as follows: 
+```js
+namePart = new ChildNodePart(name)
+emailPart = new ChildNodePart(link)
+emailAttributePart = new AttributePart(link, ’href’)
```

Comment:
it looks single quotes are unintentionally converted to smart quotes throughout the draft.